provider/cluster_rosa_classic_resource.go

```diff
@@ -1201,6 +1285,129 @@ func (r *ClusterRosaClassicResource) Update(ctx context.Context, request tfsdk.U
 	response.Diagnostics.Append(diags...)
 }
 
+// Upgrades the cluster if the desired (plan) version is greater than the
+// current version
+func (r *ClusterRosaClassicResource) upgradeClusterIfNeeded(ctx context.Context, state, plan *ClusterRosaClassicState) error {
+	if common.IsStringAttributeEmpty(plan.Version) || common.IsStringAttributeEmpty(state.CurrentVersion) {
+		// No version information, nothing to do
+		r.logger.Debug(ctx, "Insufficient cluster version information to determine if upgrade should be performed.")
+		return nil
+	}
```

Comment:
I believe both can happen legitimately.
- Users are not required to specify a version (they get the default at creation)... Subsequent applies would then not have a value here.
- If the cluster hasn't finished building, current_version isn't yet available (see `rosa describe cluster` output)
I don't think either are an error.

Reply:
Ok, this line is called also in first create, this is why it possible
What do you say about spliting this if statement?

- First checks `common.IsStringAttributeEmpty(state.CurrentVersion)`, if it empty so we know for sure its first creation (not update) so we can return (print info log, also mention this is not expected to be upgrade)
- Second check - `common.IsStringAttributeEmpty(plan.Version)` - If the version is also calculated by the provider, so it expected to be not empty, in any not first create call, am I right? so in this case should return error?

Reply:
OCM does not have the current cluster version for some time after creation (I don't know why) `rosa describe cluster` has a blank version for a while, so it's not a TF issue. So, we do need this check.
Unless the user specifically sets `version` in their config file, it will be blank. It's not set by the provider based on a read from the cluster (i.e., it's not `calculated`... only `current_version` is).

Reply:
OK